### ISSUE TITLE
ENAMETOOLONG: name too long when offlining th.wikipedia.org

### ISSUE THREAD

**Issue by automactic** [openzim/mwoffliner#555]
Command run: `mwoffliner --mwUrl=https://th.wikipedia.org --adminEmail=contact@kiwix.org --format=nopic --format=novid --withZimFullTextIndex --redis=redis://redis --outputDirectory=/output`

Exit code: 2

```
[error] [2019-02-07T21:27:34.929Z] Error: Failed to parse JSON response: [[object Object]]
at resps.reduce.pages (/usr/lib/node_modules/mwoffliner/lib/mwoffliner.lib.js:2116:59)
at Array.reduce (<anonymous>)
at /usr/lib/node_modules/mwoffliner/lib/mwoffliner.lib.js:2111:58
at process._tickCallback (internal/process/next_tick.js:68:7)
[error] [2019-02-07T21:27:35.117Z] Error: Failed to parse JSON response: [[object Object]]
at resps.reduce.pages (/usr/lib/node_modules/mwoffliner/lib/mwoffliner.lib.js:2116:59)
at Array.reduce (<anonymous>)
at /usr/lib/node_modules/mwoffliner/lib/mwoffliner.lib.js:2111:58
at process._tickCallback (internal/process/next_tick.js:68:7)
[error] [2019-02-07T21:27:35.304Z] Error: Failed to parse JSON response: [[object Object]]
at resps.reduce.pages (/usr/lib/node_modules/mwoffliner/lib/mwoffliner.lib.js:2116:59)
at Array.reduce (<anonymous>)
at /usr/lib/node_modules/mwoffliner/lib/mwoffliner.lib.js:2111:58
at process._tickCallback (internal/process/next_tick.js:68:7)
[error] [2019-02-07T21:27:35.508Z] Error: Failed to parse JSON response: [[object Object]]
at resps.reduce.pages (/usr/lib/node_modules/mwoffliner/lib/mwoffliner.lib.js:2116:59)
at Array.reduce (<anonymous>)
at /usr/lib/node_modules/mwoffliner/lib/mwoffliner.lib.js:2111:58
at process._tickCallback (internal/process/next_tick.js:68:7)
[error] [2019-02-07T21:27:35.727Z] Error: Failed to parse JSON response: [[object Object]]
at resps.reduce.pages (/usr/lib/node_modules/mwoffliner/lib/mwoffliner.lib.js:2116:59)
at Array.reduce (<anonymous>)
at /usr/lib/node_modules/mwoffliner/lib/mwoffliner.lib.js:2111:58
at process._tickCallback (internal/process/next_tick.js:68:7)
[error] [2019-02-07T21:27:35.929Z] Error: Failed to parse JSON response: [[object Object]]
at resps.reduce.pages (/usr/lib/node_modules/mwoffliner/lib/mwoffliner.lib.js:2116:59)
at Array.reduce (<anonymous>)
at /usr/lib/node_modules/mwoffliner/lib/mwoffliner.lib.js:2111:58
at process._tickCallback (internal/process/next_tick.js:68:7)
[error] [2019-02-07T21:27:36.132Z] Error: Failed to parse JSON response: [[object Object]]
at resps.reduce.pages (/usr/lib/node_modules/mwoffliner/lib/mwoffliner.lib.js:2116:59)
at Array.reduce (<anonymous>)
at /usr/lib/node_modules/mwoffliner/lib/mwoffliner.lib.js:2111:58
at process._tickCallback (internal/process/next_tick.js:68:7)
[error] [2019-02-07T21:27:36.314Z] Error: Failed to parse JSON response: [[object Object]]
at resps.reduce.pages (/usr/lib/node_modules/mwoffliner/lib/mwoffliner.lib.js:2116:59)
at Array.reduce (<anonymous>)
at /usr/lib/node_modules/mwoffliner/lib/mwoffliner.lib.js:2111:58
at process._tickCallback (internal/process/next_tick.js:68:7)
[error] [2019-02-07T21:27:36.506Z] Error: Failed to parse JSON response: [[object Object]]
at resps.reduce.pages (/usr/lib/node_modules/mwoffliner/lib/mwoffliner.lib.js:2116:59)
at Array.reduce (<anonymous>)
at /usr/lib/node_modules/mwoffliner/lib/mwoffliner.lib.js:2111:58
at process._tickCallback (internal/process/next_tick.js:68:7)
[error] [2019-02-07T21:27:36.708Z] Error: Failed to parse JSON response: [[object Object]]
at resps.reduce.pages (/usr/lib/node_modules/mwoffliner/lib/mwoffliner.lib.js:2116:59)
at Array.reduce (<anonymous>)
at /usr/lib/node_modules/mwoffliner/lib/mwoffliner.lib.js:2111:58
at process._tickCallback (internal/process/next_tick.js:68:7)
[error] [2019-02-07T21:27:36.911Z] Error: Failed to parse JSON response: [[object Object]]
at resps.reduce.pages (/usr/lib/node_modules/mwoffliner/lib/mwoffliner.lib.js:2116:59)
at Array.reduce (<anonymous>)
at /usr/lib/node_modules/mwoffliner/lib/mwoffliner.lib.js:2111:58
at process._tickCallback (internal/process/next_tick.js:68:7)
Couldn\'t find strings file for [th], falling back to [en]
[error] [2019-02-07T21:31:54.429Z] Absolutely unable to retrieve async. URL: http://upload.wikimedia.org/wikipedia/en/a/ae/MP-magnifying-glass.png
[error] [2019-02-07T21:31:54.433Z] Absolutely unable to retrieve async. URL: http://upload.wikimedia.org/wikipedia/en/thumb/2/20/Tango-video-x-generic.png/40px-Tango-video-x-generic.png
[error] [2019-02-07T21:31:54.449Z] Absolutely unable to retrieve async. URL: http://upload.wikimedia.org/wikipedia/en/7/7e/MP-open-book.png
[error] [2019-02-07T21:31:54.453Z] Absolutely unable to retrieve async. URL: http://upload.wikimedia.org/wikipedia/en/9/9f/MP-three-books.png
[error] [2019-02-07T21:31:54.460Z] Absolutely unable to retrieve async. URL: http://upload.wikimedia.org/wikipedia/en/a/a7/Monobook-globe.png
[error] [2019-02-07T21:31:54.697Z] Absolutely unable to retrieve async. URL: https://th.wikipedia.org/wiki/Mediawiki:offline.css?action=raw
Failed to run mwoffliner after [494s]: {
\t"message": "Fatal Error:",
\t"error": {
\t\t"stack": "Error: ENAMETOOLONG: name too long, open \'/tmp/wikipedia_th_all_nopic_2019-02/ICD-10_\xe0\xb8\x9a\xe0\xb8\x97\xe0\xb8\x97\xe0\xb8\xb5\xe0\xb9\x88_18:_\xe0\xb8\xad\xe0\xb8\xb2\xe0\xb8\x81\xe0\xb8\xb2\xe0\xb8\xa3_\xe0\xb8\xad\xe0\xb8\xb2\xe0\xb8\x81\xe0\xb8\xb2\xe0\xb8\xa3\xe0\xb9\x81\xe0\xb8\xaa\xe0\xb8\x94\xe0\xb8\x87_\xe0\xb9\x81\xe0\xb8\xa5\xe0\xb8\xb0\xe0\xb8\x84\xe0\xb8\xa7\xe0\xb8\xb2\xe0\xb8\xa1\xe0\xb8\x9c\xe0\xb8\xb4\xe0\xb8\x94\xe0\xb8\x9b\xe0\xb8\x81\xe0\xb8\x95\xe0\xb8\xb4\xe0\xb8\x97\xe0\xb8\xb5\xe0\xb9\x88\xe0\xb8\x9e\xe0\xb8\x9a\xe0\xb8\x88\xe0\xb8\xb2\xe0\xb8\x81\xe0\xb8\x81\xe0\xb8\xb2\xe0\xb8\xa3\xe0\xb8\x95\xe0\xb8\xa3\xe0\xb8\xa7\xe0\xb8\x88\xe0\xb8\x97\xe0\xb8\xb2\xe0\xb8\x87\xe0\xb8\x84\xe0\xb8\xa5\xe0\xb8\xb4\xe0\xb8\x99\xe0\xb8\xb4\xe0\xb8\x81\xe0\xb9\x81\xe0\xb8\xa5\xe0\xb8\xb0\xe0\xb8\x97\xe0\xb8\xb2\xe0\xb8\x87\xe0\xb8\xab\xe0\xb9\x89\xe0\xb8\xad\xe0\xb8\x87\xe0\xb8\x9b\xe0\xb8\x8f\xe0\xb8\xb4\xe0\xb8\x9a\xe0\xb8\xb1\xe0\xb8\x95\xe0\xb8\xb4\xe0\xb8\x81\xe0\xb8\xb2\xe0\xb8\xa3.html\'",
\t\t"message": "ENAMETOOLONG: name too long, open \'/tmp/wikipedia_th_all_nopic_2019-02/ICD-10_\xe0\xb8\x9a\xe0\xb8\x97\xe0\xb8\x97\xe0\xb8\xb5\xe0\xb9\x88_18:_\xe0\xb8\xad\xe0\xb8\xb2\xe0\xb8\x81\xe0\xb8\xb2\xe0\xb8\xa3_\xe0\xb8\xad\xe0\xb8\xb2\xe0\xb8\x81\xe0\xb8\xb2\xe0\xb8\xa3\xe0\xb9\x81\xe0\xb8\xaa\xe0\xb8\x94\xe0\xb8\x87_\xe0\xb9\x81\xe0\xb8\xa5\xe0\xb8\xb0\xe0\xb8\x84\xe0\xb8\xa7\xe0\xb8\xb2\xe0\xb8\xa1\xe0\xb8\x9c\xe0\xb8\xb4\xe0\xb8\x94\xe0\xb8\x9b\xe0\xb8\x81\xe0\xb8\x95\xe0\xb8\xb4\xe0\xb8\x97\xe0\xb8\xb5\xe0\xb9\x88\xe0\xb8\x9e\xe0\xb8\x9a\xe0\xb8\x88\xe0\xb8\xb2\xe0\xb8\x81\xe0\xb8\x81\xe0\xb8\xb2\xe0\xb8\xa3\xe0\xb8\x95\xe0\xb8\xa3\xe0\xb8\xa7\xe0\xb8\x88\xe0\xb8\x97\xe0\xb8\xb2\xe0\xb8\x87\xe0\xb8\x84\xe0\xb8\xa5\xe0\xb8\xb4\xe0\xb8\x99\xe0\xb8\xb4\xe0\xb8\x81\xe0\xb9\x81\xe0\xb8\xa5\xe0\xb8\xb0\xe0\xb8\x97\xe0\xb8\xb2\xe0\xb8\x87\xe0\xb8\xab\xe0\xb9\x89\xe0\xb8\xad\xe0\xb8\x87\xe0\xb8\x9b\xe0\xb8\x8f\xe0\xb8\xb4\xe0\xb8\x9a\xe0\xb8\xb1\xe0\xb8\x95\xe0\xb8\xb4\xe0\xb8\x81\xe0\xb8\xb2\xe0\xb8\xa3.html\'",
\t\t"errno": -36,
\t\t"code": "ENAMETOOLONG",
\t\t"syscall": "open",
\t\t"path": "/tmp/wikipedia_th_all_nopic_2019-02/ICD-10_\xe0\xb8\x9a\xe0\xb8\x97\xe0\xb8\x97\xe0\xb8\xb5\xe0\xb9\x88_18:_\xe0\xb8\xad\xe0\xb8\xb2\xe0\xb8\x81\xe0\xb8\xb2\xe0\xb8\xa3_\xe0\xb8\xad\xe0\xb8\xb2\xe0\xb8\x81\xe0\xb8\xb2\xe0\xb8\xa3\xe0\xb9\x81\xe0\xb8\xaa\xe0\xb8\x94\xe0\xb8\x87_\xe0\xb9\x81\xe0\xb8\xa5\xe0\xb8\xb0\xe0\xb8\x84\xe0\xb8\xa7\xe0\xb8\xb2\xe0\xb8\xa1\xe0\xb8\x9c\xe0\xb8\xb4\xe0\xb8\x94\xe0\xb8\x9b\xe0\xb8\x81\xe0\xb8\x95\xe0\xb8\xb4\xe0\xb8\x97\xe0\xb8\xb5\xe0\xb9\x88\xe0\xb8\x9e\xe0\xb8\x9a\xe0\xb8\x88\xe0\xb8\xb2\xe0\xb8\x81\xe0\xb8\x81\xe0\xb8\xb2\xe0\xb8\xa3\xe0\xb8\x95\xe0\xb8\xa3\xe0\xb8\xa7\xe0\xb8\x88\xe0\xb8\x97\xe0\xb8\xb2\xe0\xb8\x87\xe0\xb8\x84\xe0\xb8\xa5\xe0\xb8\xb4\xe0\xb8\x99\xe0\xb8\xb4\xe0\xb8\x81\xe0\xb9\x81\xe0\xb8\xa5\xe0\xb8\xb0\xe0\xb8\x97\xe0\xb8\xb2\xe0\xb8\x87\xe0\xb8\xab\xe0\xb9\x89\xe0\xb8\xad\xe0\xb8\x87\xe0\xb8\x9b\xe0\xb8\x8f\xe0\xb8\xb4\xe0\xb8\x9a\xe0\xb8\xb1\xe0\xb8\x95\xe0\xb8\xb4\xe0\xb8\x81\xe0\xb8\xb2\xe0\xb8\xa3.html"
\t
```

**Comment by kelson42** [comment]
@ISNIT0 I can reproduce the problem with 1.7. Really interested to know what is going one here, as this web site is basically the same as the other Wikipedias.

**Comment by kelson42** [comment]
@ISNIT0 Maybe this is because the size of the filename is limited on the fs?

**Comment by ISNIT0** [comment]
Have we always mirrored the `th` wikipedia?
I guess it is a filesystem limitation

**Comment by kelson42** [comment]
@ISNIT0 Yes (see https://download.kiwix.org/zim/wikipedia/) and this is the strange thing. Please check we do not have a regression here.

**Comment by ISNIT0** [comment]
@kelson42 This will hopefully not be an issue going forwards because of #278

**Comment by kelson42** [comment]
@ISNIT0 Does `mwoffliner` 1.8 suffer finally of the same problem?

**Comment by ISNIT0** [comment]
Fixed in `1.8`
